### ISSUE TITLE
Hard drives can not be dormant

### ISSUE THREAD

**Issue by lucktu** [ntop/n2n#189]
Yes, n2n_v2 and meyerd's n2n are sure to be this way without communication, but v1 (or tinc) is normal.

edge1 -d D1 -a 172.1.0.100 -c test1 -k 123 -l xxx.com:10082 -br   #v1
edge2 -d D2 -a 172.2.0.100 -c test2 -k 234 -l xxx.com:10086 -rEfA #v2

**Comment by lucktu** [comment]
edge -a 172.10.0.100 -c n2n -k test -l xxxx.xxx:10086 -b

**Comment by emanuele-f** [comment]
Please explain in more detail the issue, I don't understand what you mean

**Comment by lucktu** [comment]
I have an synology(nas) with a simplify Linux. When it is hibernation, the light of hard disk will go out. When it wake up, it will be recorded in log(as shown). Based on this, I can tell if the machine is hibernation.
![image](https://github.com/lucktu/other/raw/master/image/speed/19110301.png)

in rc.local，add n2n(v1+v2+v2s) frpc tinc
edge1 -d D1 -a 172.1.0.100 -c test1 -k 123 -l xxx.com:10082 -br    #v1
edge2 -d D2 -a 172.2.0.100 -c test2 -k 234 -l xxx.com:10086 -rEfA #v2
edge2s -d D3 -a 172.3.0.100 -c test3 -k 342 -l xxx.com:10088 -rEf  #v2s
... ...

      08:31 -- 09:20 frpc + tinc + edge(v1+v2+v2s): after rebooting synology, not hibernation
      09:32 -- 10:02 frpc + edge(v1+v2+v2s):  killall tinc , not hibernation
      10:03 -- 10:15 frpc: killall edge , to hibernation
      10:42 -- 10:55 frpc + edge_v1(run edge_v1 manually): to hibernation
      10:58 -- 11:18 frpc + edge(v1+v2+v2s): run edge_v2 & v2s manually, not hibernation
      11:19 -- 11:40 frpc + edge(v1+v2s): killall edge_v2, not hibernation
      11:40 -- 11:52 frpc + edge(v1): killall edge_v2s, to hibernation
      11:54 -- 12:xx frpc + tinc + edge(v1): run tinc manually, to hibernation
      12:19 -- 12:xx modify rc.local, reboot synology, to hibernation
      12:41 -- 13:12 frpc + tinc + edge(v1+v2): run edge_v2 manually, not hibernation
      13:53 -- 14:xx frpc + tinc + edge(v1): killall edge_v2, to hibernation

v2s -- meyerd's n2n

**Comment by emanuele-f** [comment]
So the issue you are experiencing is that while n2n is running your NAS disk won't hibernate for power saving? n2n should not perform any disk access as far as I know

**Comment by lucktu** [comment]
I have ascertained that the problem is due to the "E" parameter.

**Comment by lucktu** [comment]
> 
> 
> So the issue you are experiencing is that while n2n is running your NAS disk won't hibernate for power saving? n2n should not perform any disk access as far as I know

I am sorry, yes, it is.

**Comment by emanuele-f** [comment]
You can easily monitor `write` syscalls with `sudo strace -e write edge ...your_edge_options...`. You can do the same for `read` syscalls. Maybe you are logging the n2n edge output to file?

**Comment by lucktu** [comment]
That's a tough question to answer. today I tested the last command line again and found that the computer was dormant. but I think the parameter E factor is bigger (without E, the computer is always dormant), especially when running multiple edges, when the network is complex. 

here are some tests I did (In 7 hours, over and over again). (OK: can hibernate), 
<pre>
edge2 -d D2 -a 172.2.0.100 -c test2 -k 234 -l xxx.com:10086 -rEfA   #  OK
edge2 -d D2 -a 172.2.0.100 -c test2 -k 234 -l xxx.com:10086 -p 21282 -t 56232 -ErfA #  OK
edge2 -d D2 -a 172.2.0.100 -c test2 -k 234 -l xxx.com:10086 -p 21282 -t 56232 -ErfA > /tmp/apple/WE.txt #  NO
edge2 -d D2 -a 172.2.0.100 -c test2 -k 234 -l xxx.com:10086 -p 30002 -t 50002 -ErfA > /tmp/apple/WE.txt #  OK
edge2 -d D2 -a 172.2.0.100 -c test2 -k 234 -l xxx.com:10086 -p 21282 -t 56232  -rfA > /tmp/apple/WO.txt #  OK
</pre>

Same thing, not necessarily repeatable.  they use two sumernodes.

With and without E, the output is the same:
<pre>
strace -e write edge2 -d D2 -a 172.2.0.100 -c test2 -k 234 -l xxx.com:10086 -p 21282 -t 56232 -ErfA > /tmp/apple/WE.txt &
</pre>

So, This post can only be closed, I am sorry!
